response.go

```diff
@@ -70,19 +81,19 @@ func (err connectionError) Timeout() bool {
 	return true
 }
 
-// serverError represents internal server errors.
+// ServerError represents internal server errors.
 // Implements `net.Error` interface.
-type serverError string
+type ServerError string
```

Comment:
It is non-idiomatic to declare a type that has exported func members, but the type itself is not exported. It makes it difficult to do things like this:

```go
result, err := someReallyDeepNestedCall()
if err != nil {
  switch v := err.(type) {
  case *fcm.ServerError:
  case *otherpackage.Error:
  }
}
```

I am not suggesting that the above is a "good" way to write error handling, but without exporting the ServerError type on fcm, you can't do effective / idiomatic type assertions. 

Typically, any thing you return (that's not just a plain "error" from something like `errors.New()`) should be an exported type.